nucleus/hk2/hk2-config/src/main/java/org/jvnet/hk2/config/ConfigParser.java

```diff
@@ -74,16 +74,27 @@
  * @author Kohsuke Kawaguchi
  */
 public class ConfigParser {
+
+    private static final Logger LOGGER = Logger.getAnonymousLogger();
+
     /**
      * This is where we put parsed inhabitants into.
      */
     protected final ServiceLocator habitat;
 
+    private final List<String> errors;
+
+    private boolean logErrors;
 
     public ConfigParser(ServiceLocator habitat) {
-        this.habitat = habitat;
+        this(habitat, true);
     }
 
+    public ConfigParser(ServiceLocator habitat, boolean logErrors) {
+        this.habitat = habitat;
+        this.logErrors = logErrors;
+        this.errors = new ArrayList<>();
```

Comment:
Somewhere in javadoc should be said that the parser is not reusable (errors have state before and after first use)